### ISSUE TITLE
Feature request: Add "Rebase <x> branch on top of this"

### ISSUE THREAD

**Issue by ruggi99** [desktop/desktop#15282]
### The feature request

In the Branch Menu there's "Rebase current branch" entry that rebases the current branch onto the selected one in the dialog.
If I want to merge a feature branch and I want a linear commit history and no squashing, I have to switch to the feature branch, rebase it on top of the default branch, switch back to the default branch and then finally merge that branch (doing a fast-forward)

### Proposed solution

What I would like to have is the inverse of "Rebase current branch", so rebase the selected branch (in the dialog) onto the current branch, but without actually swithing branch manually.
It's like "Squash and merge into current" but rebase instead of squash.
This would enable a workflow that is similar of when merging PRs in GitHub but with "rebase" option (the third), done locally.

Thanks in advance

**Comment by ruggi99** [comment]
Any thoughts on this?

**Comment by niik** [comment]
@ruggi99 I _think_ (though it's early in the morning and I haven't had my coffee yet) that what you want is to use the merge dialog (Branch -> Merge into current branch) and then use the merge option at the bottom of the dialog to say that you want to rebase rather than merge

<img width="472" alt="image" src="https://user-images.githubusercontent.com/634063/191185050-0d80de2e-bfa1-46af-9181-aee1e7b55042.png">

Does that make sense?

**Comment by ruggi99** [comment]
Hi @niik.
Unfortunately it does not work like you said.
When I do a rebase following your steps, what I get is main (or master), that is the current selected branch, is rebased onto the feature branch I want to rebase.
What I would like is the inverse, so rebase the feature branch on top of main.

Let me better explain me:
Initially I have this situation:
![image](https://user-images.githubusercontent.com/25064808/191189622-8fce0ac2-ae78-4697-8a78-fb70cbf742eb.png)


https://user-images.githubusercontent.com/25064808/191190421-6d475393-2651-45f5-bbb0-62fa421d92d1.mp4

Like you can see in the video, the last commit in main is placed above the last commit in branch test.
What I would like is the inverse, so "Update README.md" above the last commit in main.
Don't know if I explained better.

**Comment by niik** [comment]
Hmm, yeah, that doesn't seem like it matches the description in the dialog of what it should do 

<img width="454" alt="image" src="https://user-images.githubusercontent.com/634063/191191998-1abac0ff-2807-486c-a4df-38d0b8bbea4f.png">

And with this behavior I'm not sure what the point would be in having a separate rebase dialog. It was my understanding that you'd use the rebase dialog to rebase your current branch on top of another but you could use the merge dialog's rebase mode to rebase another branch _on top of the current branch_. Let me do some digging to see what it is that I'm missing here

**Comment by ruggi99** [comment]
Yeah, like you said, the description in the dialog is what I would to do, but not how it works now

**Comment by ruggi99** [comment]
@niik  Any update to this?

**Comment by niik** [comment]
@ruggi99 Hey, yeah sorry for not reporting back sooner. We talked about this and discussed it with friends working on Git itself. There's work in progress within Git to support this in a more streamlined fashion. As it stands today in order to implement this we'd have to first create a temporary branch based on the feature branch, then rebase that on top of main, switch back to main and do a fast forward merge. There's too many moving parts in this for us to feel comfortable so we'll hold off until Git gains the capability to do this in a better way.

In the meantime however we should definitely change the description here to match what it's actually doing.

**Comment by ruggi99** [comment]
Agree on both sections.

Thanks

**Comment by ruggi99** [comment]
@niik any update on this?
Has git supported this feature?

**Comment by Andrej730** [comment]
Would love option to rebase from UI in any way, 99% pulls for me is going to git console and typing `git pull --rebase`.

**Comment by steveward** [comment]
@Andrej730 if you always want to pull on rebase you can configure Git to always do it:

`git config --global pull.rebase true` 

You can also use the `--local` flag if you'd prefer it on a per-repo basis. GitHub Desktop will respect this setting.

**Comment by Andrej730** [comment]
@steveward it worked, thank you!
